File: Minimum_Number_of_Operations_to_Make_Array_Empty.java
class Solution {
    public int minOperations(int[] nums) {
        HashMap<Integer, Integer> freq = new HashMap<>();

        for (int n : nums) {
            freq.put(n, 1 + freq.getOrDefault(n, 0));
        }

        if (freq.containsValue(1)) {
            return -1;
        }

        return freq.values().stream().mapToInt(f -> (f + 2) / 3).sum();        
    }
}

File: Minimum_Number_of_Operations_to_Make_Array_Empty.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        freq = {}
        
        for n in nums:
            freq[n] = 1 + freq.get(n, 0)
        
        if 1 in freq.values():
            return -1
        
        return sum((f + 2) // 3 for f in freq.values())
